### DISCUSSION TITLE
receiving runner shutdown signal while executing codeql in GitHub Actions

### DISCUSSION THREAD

**Discussion by PoleMadhu123** [community/community#63120] [Code Security] [answered]
### Select Topic Area

Question

### Body

While Executing the CodeQL in GitHub actions, it's throwing error like this " The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled." we are using ubuntu-latest OS.

This are the commads initialize CodeQL & perform codeql steps: - 
        name: Initialize CodeQL
        uses: github/codeql-action/init@v2
        with:
          languages: javascript  
          ram: 9500  
          
     - name: Perform CodeQL Analysis
        uses: github/codeql-action/analyze@v2
   
please suggest the solutions for this error.

**Comment by jketema** [comment]
Hi @PoleMadhu123,

Thanks for you question. Is this with GitHub hosted runners, or with self-hosted ones?

**Reply by PoleMadhu123** [reply]
Hi @jketema 
Thanks for the response. we are using GitHub hosted runner.

**Reply by jketema** [reply]
Is this on a public or a private repository. If it's public, can you share a link to it?

**Reply by PoleMadhu123** [reply]
no, it's a private repository.

**Reply by jketema** [reply]
I've not seen this problem happening before with CodeQL jobs. It would help to know what the workflow is outputting just before the shutdown signal occurs. One possibility is that you've hit a bug somewhere in out code that causes it to enter an infinite loop, but that's very difficult to tell based on the information you provided.

If the private repo is part of an organisation that is GitHub customer, I'd suggest going through the official support channels, because the support engineers will have tools available that might better help determine what is going on.

**Reply by PoleMadhu123** [reply]
issue has been resolved now. we have updated the build in codeql pipeline. previously we were using "npm run build" command.
Now we are using codeql auto build command.
Thanks for your response. @jketema 
and please share links of the official support channels. it will be helpful to me.

**Reply by jketema** [reply, marked as answer]
> previously we were using "npm run build" command.

That will never have worked, and that might just have crashed something later in the pipeline. For dynamic languages like JavaScript the autobuild command is what always should be used.

> and please share links of the official support channels. it will be helpful to me.

For contacting support see here: https://docs.github.com/en/support/contacting-github-support. This assumes you're a paying customer.

**Reply by PoleMadhu123** [reply]
Thanks for the valuable information!

**Reply by PoleMadhu123** [reply]
@jketema, I have few more doubts related to codeql alerts.
1. some alerts showing "PREVIEW UNAVAILABLE", " SORRY WE COULDN'T FIND THIS FILE IN THE REPOSITORY". what does that mean, can we ignore those alerts or what?
2. what is "CWE under weakness" and how to resolve those?

**Comment by jketema** [comment]
> 1. some alerts showing "PREVIEW UNAVAILABLE", " SORRY WE COULDN'T FIND THIS FILE IN THE REPOSITORY". what does that mean, can we ignore those alerts or what?

Those will either be coming from generated files, or from dependencies you pulled in. The file name should give you a clue about which of these it is. I cannot in general say if you can ignore the alert. It very much depends on the alert and the file it is coming from.

> 2. what is "CWE under weakness" and how to resolve those?

I've not seen that before. Could you share a screenshot? Feel free to blur out any sensitive details.

**Reply by jketema** [reply]
In the screenshot no preview is available, because the file is generated (as is says just after the file name). The alert you're getting "Creating an ASP.NET debug binary may reveal sensitive information" is apparently covered by two CWEs: [11](https://cwe.mitre.org/data/definitions/11.html) and [532](https://cwe.mitre.org/data/definitions/532.html).

**Reply by jketema** [reply]
The description and recommendation should provide enough information in this case to resolve or ignore the alert.

**Reply by PoleMadhu123** [reply]
![image](https://github.com/community/community/assets/114417100/653a3cb2-3a5a-41b1-9357-887496becf2a)

**Reply by jketema** [reply]
Yes, that's one example of what might be going on and how to fix it. Note that this is just an example, the cause and solution might be different in your case.

**Reply by PoleMadhu123** [reply]
Okay @jketema , Thanks for the information.
